cloud/migrate-mst-cloud/schema-then-data.md

```diff
@@ -0,0 +1,352 @@
+    permissions errors, include the `--no-owner` flag:
+    ```bash
+    pg_restore -U tsdbadmin -W \
+    -h <cloud_host> -p <cloud_port> --no-owner -Fc \
+    -v -d tsdb <file-name-pre-data>.bak
+    ```
+
+</procedure>
+
+### Troubleshooting
+If you see any of these errors during the migration process, you can safely
+ignore them. The migration still occurs successfully.
+
+`pg_restore` tries to apply the TimescaleDB extension when it copies your
+schema. This can cause a permissions error. Because TimescaleDB is already
+installed by default on Timescale Cloud, you can safely ignore this.
+
+```bash
+pg_restore: creating EXTENSION "timescaledb"
+pg_restore: creating COMMENT "EXTENSION timescaledb"
+pg_restore: while PROCESSING TOC:
+pg_restore: from TOC entry 6239; 0 0 COMMENT EXTENSION timescaledb
+pg_restore: error: could not execute query: ERROR:  must be owner of extension timescaledb
+```
+
+If you have continuous aggregates, you might get the following error. Ignore
+this, because you [restore your aggregates](#recreate-continuous-aggregates)
+later on.
+
+```bash
+pg_restore: error: could not execute query: ERROR:  relation "_timescaledb_internal._materialized_hypertable_x" does not exist
+```
+
+```bash
+​​pg_restore: WARNING:  no privileges were granted for "<..>"
+```
+
+## Restore hypertables in Timescale Cloud
+After pre-data migration, your hypertables from Managed Service for TimescaleDB
+become regular PostgreSQL tables in Cloud. Recreate your hypertables in Cloud to
+restore them.
+
+<procedure>
+
+### Restoring hypertables in Timescale Cloud
+1.  Connect to your Timescale Cloud database:
+    ```sql
+    psql "postgres://tsdbadmin:<cloud_password>@<cloud_host>:<cloud_port>/tsdb?sslmode=require"
+    ```
+1.  Restore the hypertable:
+    ```sql
+    SELECT create_hypertable(
+       '<table_name>', '<time_column_name>',
+        chunk_time_interval =>
+            INTERVAL '<chunk_time_interval>');
+    ```
+
+</procedure>
+
+## Copy data from Managed Service for TimescaleDB
+After restoring your hypertables, return to Managed Service for TimescaleDB to
+copy your data, table by table.
+
+<procedure>
+
+### Copying data from Managed Service for TimescaleDB
+1.  Connect to your Managed Service for TimescaleDB database:
+    ```bash
+    psql "postgres://tsdbadmin:<mst_password>@<mst_host>:<mst_port>/tsdb?sslmode=require"
+    ```
+1.  Dump the data from the first table into a `.csv` file:
+    ```sql
+    COPY (SELECT * FROM <table-name>) TO <table-name>.csv CSV
+    ```
+    Repeat for each table and hypertable you want to migrate.
+
+</procedure>
+
+<highlight type="note">
+If your tables are very large, you can migrate each table in multiple pieces.
+Split each table by time range, and copy each range individually. For example:
+
+```sql
+COPY (SELECT * FROM <table-name> WHERE time > ‘2021-11-01’ AND time < ‘2011-11-02’) TO <table-name-date-range>.csv CSV
```

Comment:
```suggestion
\COPY (SELECT * FROM <table-name> WHERE time > ‘2021-11-01’ AND time < ‘2011-11-02’) TO <table-name-date-range>.csv CSV
```

Reply:
`\COPY` is used when running postgres copy from a client like psql
If you run copy server side (e.g. like pg_dump) you can use `COPY`